src/media/class-wpml-elementor-media-nodes-iterator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+class WPML_Elementor_Media_Nodes_Iterator implements IWPML_PB_Media_Nodes_Iterator {
+
+	/** @var WPML_Elementor_Media_Node_Provider $node_provider */
+	private $node_provider;
+
+	public function __construct( WPML_Elementor_Media_Node_Provider $node_provider ) {
+		$this->node_provider = $node_provider;
+	}
+
+	/**
+	 * @param array $data_array
+	 * @param string $lang
+	 * @param string $source_lang
+	 *
+	 * @return array
+	 */
+	public function translate( $data_array, $lang, $source_lang ) {
+		foreach ( $data_array as &$node ) {
+			if ( isset( $node['elements'] ) && $node['elements'] ) {
+				$node['elements'] = $this->translate( $node['elements'], $lang, $source_lang );
+			} elseif ( isset( $node['elType'], $node['widgetType'], $node['settings'] )
+			           && 'widget' === $node['elType']
+			) {
```

Comment:
@strategio perhaps you can extract the two expressions into two private methods:
- `hasElements` (the "$node" argument will give the context)
- `haselementType`

The second one, in particular, is a bit too long. The first one could be extracted for consistency.

Reply:
Makes sense @andreasciamanna, extracted!